service/k8sclient/kubectl/kubectl.go

```diff
@@ -40,24 +48,106 @@ type KubeCtl struct {
 
 // NewKubeCtl creates a new KubeCtl object with a given logger.
 func NewKubeCtl(l logger.Logger) *KubeCtl {
-	// TODO Handle kubectl versions https://jira.percona.com/browse/PMM-6348
-
-	// TODO Handle kubeconfig https://jira.percona.com/browse/PMM-6347
-
-	cmd := []string{"dbaas-kubectl-1.16"}
-	path, err := exec.LookPath(cmd[0])
-	l.Debugf("path: %s, err: %v", path, err)
-	if e, ok := err.(*exec.Error); err != nil && ok && e.Err == exec.ErrNotFound {
-		cmd = []string{"minikube", "kubectl", "--"}
+	// Handle kubectl versions
+	cmd, err := getKubectlCmd()
+	if err != nil {
+		l.Debugf("Cannot find kubectl binary: %s", err)
+		return nil
 	}
 	l.Debugf("Using %q", strings.Join(cmd, " "))
 
+	// TODO Handle kubeconfig https://jira.percona.com/browse/PMM-6347
+
 	return &KubeCtl{
 		l:   l.WithField("component", "kubectl"),
 		cmd: cmd,
 	}
 }
 
+// getKubectlCmd gets correct version of kubectl binary for Kubernetes cluster.
+func getKubectlCmd() ([]string, error) {
+	// Firstly lookup default kubectl to get Kubernetes Server version.
+	kubectlCmd, err := lookupCorrectKubectlCmd([]string{defaultKubectl})
+	if err != nil {
+		return nil, err
+	}
+
+	versionsJSON, err := getVersions(kubectlCmd)
+	if err != nil {
+		return nil, err
+	}
+
+	kubectlCmdNames, err := selectCorrectKubectlVersions(versionsJSON)
+	if err != nil {
+		return nil, err
+	}
+
+	return lookupCorrectKubectlCmd(kubectlCmdNames)
+}
+
+// getVersions gets kubectl and Kubernetes cluster version.
+func getVersions(kubectlCmd []string) ([]byte, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+
+	getVersionCmd := append(kubectlCmd, "version", "-o", "json")
+	cmd := exec.CommandContext(ctx, getVersionCmd[0], getVersionCmd[1:]...) //nolint:gosec
```

Comment:
I think we should refactor `run` method to make it a function, and the use it everywhere instead of exec.Command/CommandContext

Reply:
Agree. Also thought about this.